app/dao/service_data_retention_dao.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+
+from app import db
+from app.dao.dao_utils import transactional
+from app.models import ServiceDataRetention
+
+
+def fetch_service_data_retention_by_id(service_id, data_retention_id):
```

Comment:
If you have the data_retention_id do you need the service_id?

Reply:
The service id is only there as a safety check, that the service is requesting the data for it's own data_retention_id.

Reply:
We have done this in other queries by id. Just to ensure the service is getting it's own data and not data for a different service